### PR TITLE
Run sdp-operator as non-root

### PR DESCRIPTION
**Pull request by mandopaloooza**
## Description
* Place python env and assets into non-root directory
* Create user `appgate`
* Run deployment as `appgate` user

## Checklist
- [x] Bump `.version` and `.appVersion` in [k8s/crd/Chart.yaml](../k8s/crd/Chart.yaml)
- [x] Bump `.version` and `.appVersion` in [k8s/operator/Chart.yaml](../k8s/operator/Chart.yaml)

### CODE REVIEW

File: docker/assets/run.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-/root/venv/bin/python3 -m appgate $@
+/appgate/venv/bin/python3 -m appgate $@
```
Comment (thomascellerier) [inline docker/assets/run.sh:3]: Not related directly to this PR but we could remove one extra process by using exec here:
```suggestion
exec /appgate/venv/bin/python3 -m appgate $@
```

**Review by thomascellerier — COMMENTED** [review]
Nice, some initial comments.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
+    apt-get autoremove --yes & \
+    rm -rf /var/lib/{apt,dpkg,cache,log}/ & \
+    rm /appgate/requirements.txt
+
+COPY appgate /appgate/appgate/
+COPY api_specs /appgate/api_specs
+COPY docker/assets/run.sh /appgate/run.sh
+RUN chmod +x /appgate/run.sh
+
+RUN useradd appgate --uid 1000 --user-group
```
[the remark this answers is not in the document]
Comment (mandopaloooza) [reply docker/Dockerfile:24]: Will test the git operator tomorrow

[the remark this answers is not in the document]
Comment (mandopaloooza) [reply docker/Dockerfile]: You were right. I added `--create-home` flag to allow git operations

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
+    apt-get install gcc libffi-dev git --yes && \
+    apt-get clean && \
+    apt-get autoclean
+
+COPY requirements.txt /appgate
+
+RUN python3 -m venv /appgate/venv && \
+    /appgate/venv/bin/pip install --upgrade pip && \
+    /appgate/venv/bin/pip install -r /appgate/requirements.txt && \
     apt-get autoremove gcc --yes && \
```
Comment (thomascellerier) [inline docker/Dockerfile:16]: I still think the two `RUN` instructions should be merged into one, or you will be shipping gcc in the actual image. You can move the copy before that.
Speaking of which, what do we need gcc for? Assuming we use only binary wheel python packages it shouldn't be necessary.

**Review by thomascellerier — COMMENTED** [review]
One more comment about the layers.
Else it looks good.

File: docker/Dockerfile (hunk 3 above)
Comment (mandopaloooza) [reply docker/Dockerfile:16]: Looks like it was added a while back in commit 95d5d9786640a3809c34931967d5b8480cb1c1a6. yarl didn't have python3.9 wheel yet back then. I can remove this now.

**Review by thomascellerier — APPROVED** [review]

**Pull request merged**